### PR TITLE
fixes cyborgs and monkeys not being able to buckle to chairs

### LINKED ISSUE #84047: Trying to buckle yourself to a seat as a cyborg displays this messsage
Reporting client version: 515.1639

## Issue Summary
image

## Round ID:
[230653](https://scrubby.melonmesa.com/round/230653)

## Testmerges:
- [[TM ONLY] Hijacked Revision Date for v3 TGS Api](https://www.github.com/tgstation/tgstation/pull/83227)
- [Fixes free chemsplosion reagent purging by replacing it with the same thing but dangerous](https://www.github.com/tgstation/tgstation/pull/83195)

### PR DESCRIPTION
**Pull request by grungussuss**
## About The Pull Request
closes https://github.com/tgstation/tgstation/issues/84047
## Changelog
:cl:
fix: fixed cyborgs and monkeys not being able to buckle to chairs
/:cl:

### CODE REVIEW

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	integrity_failure = 0.1
 	custom_materials = list(/datum/material/iron =SHEET_MATERIAL_AMOUNT)
 	layer = OBJ_LAYER
-	interaction_flags_mouse_drop = NEED_HANDS | NEED_DEXTERITY
+	interaction_flags_mouse_drop = NONE
```
Comment (SyncIt21) [inline code/game/objects/structures/beds_chairs/chair.dm:14]:
```suggestion
```

It's by default none. Just don't set it

**Comment by Dawnseer** [comment]
Now fix Cyborgs being unable to use modules while buckled to a chair

**Comment by Sadboysuss** [comment]
> Now fix Cyborgs being unable to use modules while buckled to a chair

make an issue report

**Comment by Dawnseer** [comment]
Are you telling me, that for the months, years, that we've had this, nobody has made an issue report?

**Comment by Sadboysuss** [comment]
> Are you telling me, that for the months, years, that we've had this, nobody has made an issue report?

it's possible, or it got lost somewhere

**Comment by Dawnseer** [comment]
I'll make an issue report after I get home from work tomorrow I guess

**Comment by Jacquerel** [comment]
Does this allow (for instance) space carp to buckle you to chairs when they could not before?

**Comment by Sadboysuss** [comment]
> Does this allow (for instance) space carp to buckle you to chairs when they could not before?

I'll check when I have the time, I think it only works if you're buckling yourself but that's a good question.

**Comment by SyncIt21** [comment]
Fixes #84218.

Even if it doesn't its the same type of fix so please look into it

**Comment by Higgin** [comment]
likely also involved in #84264 and #84267 , other behaviors probably got caught out that we don't know about yet @Sadboysuss

**Comment by SyncIt21** [comment]
> likely also involved in #84264 and #84267 , other behaviors probably got caught out that we don't know about yet @Sadboysuss

This is going way out of scope for this PR. I'll address the above 2 for now

**Comment by dalmationer** [comment]
> Fixes #84218.
> 
> Even if it doesn't its the same type of fix so please look into it

I've updated the issue report for #84218 to reflect that this appears to affect all chairs and ridable mobs, except the wheelchair.

**Comment by SyncIt21** [comment]
> Fixes #84218.
> 
> Even if it doesn't its the same type of fix so please look into it

So good news my PR fixes #84268 fixes this issue, so its not your issue anymore. As for cyborgs if you want them to buckle just remove the `NEED_HANDS` flag and you should be good

**Comment by Sadboysuss** [comment]
> > Fixes #84218.
> > 
> > Even if it doesn't its the same type of fix so please look into it
> 
> So good news my PR #84268 fixes it so a fix here is no longer required. As for cyborgs if you want them to buckle just remove the `NEED_HANDS` flag and you should be good

Problem is if I remove `NEED_HANDS` it allows mobs like carps to buckle others when they couldn't before, I thought about adding a `NEED_DEXTERITY` check but that would stop monkeys from buckling so I'm not sure what to do

**Comment by Sadboysuss** [comment]
> Does this allow (for instance) space carp to buckle you to chairs when they could not before?

basic mobs could buckle other mobs before.

**Pull request merged**